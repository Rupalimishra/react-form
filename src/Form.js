import React  from 'react';
import './Form.css';
import FormSignup from './FormSignup';
import FormSuccess from './FormSuccess';

const Form = () => {
  const [isSubmitted, setIsSubmitted] = React.useState(false);

  function submitForm() {
    setIsSubmitted(true);
  }
  return (
    <div>
      <div className='form-container'>
        {!isSubmitted ? (
          <FormSignup submitForm={submitForm} />
        ) : (
          <FormSuccess />
        )}
      </div>
    </div>
    
  );
};

export default Form;
