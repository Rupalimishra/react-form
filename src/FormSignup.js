import React from 'react';
import validate from './validateInfo';
import useForm from './useForm';
import './Form.css';

const FormSignup = ({ submitForm }) => {
  const { handleChange, handleSubmit, values, errors } = useForm(
    submitForm,
    validate
  );

  return (
    <div className='form-content-right' >
      <form onSubmit={handleSubmit} className='form' noValidate>
        <h1>
          Registration form
        </h1>
        <div className='form-inputs' >
          <label className='form-label'>Username</label>
          <input
           data-test="input-box"
            className='form-input'
            type='text'
            name='username'
            placeholder='Enter your username'
            value={values.username}
            onChange={handleChange}
          />
          {errors.username && <p>{errors.username}</p>}
        </div>
        <div className='form-inputs'>
          <label className='form-label'>Email</label>
          <input
            data-test="input-box"
            className='form-input'
            type='email'
            id='#email'
            name='email'
            placeholder='Enter your email'
            value={values.email}
            onChange={handleChange}
          />
          {errors.email && <p>{errors.email}</p>}
        </div>
        <div className='form-inputs'>
          <label className='form-label'>Password</label>
          <input
            data-test="input-box"
            className='form-input'
            type='password'
            name='password'
            placeholder='Enter your password'
            value={values.password}
            onChange={handleChange}
          />
          {errors.password && <p>{errors.password}</p>}
        </div>
        <div className='form-inputs'>
          <label className='form-label'>Confirm Password</label>
          <input
          data-test="input-box"
            className='form-input'
            type='password'
            name='password2'
            placeholder='Confirm your password'
            value={values.password2}
            onChange={handleChange}
          />
          {errors.password2 && <p>{errors.password2}</p>}
        </div>
        <div className="container">
        <button  data-test='submit button 'className='form-input-btn'   type='submit' name="submit">
          Sign up
         
        </button>
        </div>
       
      </form>
    </div>
  );
};

export default FormSignup;
