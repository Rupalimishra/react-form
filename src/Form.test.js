import { shallow } from 'enzyme';
import React, { Component } from 'react';
import Form from './Form';
import FormSignup from './FormSignup';
import FormSuccess from './FormSuccess' ;
import { findByTestAttr } from '../test/testUtils';



it("should render initial layout", () => {
    
    const component = shallow(<Form />);
    expect(component.getElements()).toMatchSnapshot();
});

describe("render the success component", () => {
    it("render FormSuccess component length", () => {
        expect(shallow(<FormSuccess />).length).toEqual(1);   
     });
     it("render the FormSignup component", () => {
         expect(shallow(<FormSignup />).length).toEqual(1);
     });
});

const setup = () => {
    return shallow(<Form />);
}

it('form render', () => {
  const wrapper = setup();
  const inputComponent = findByTestAttr(wrapper, 'form-container');
  expect(inputComponent.length).toBe(0);
})



