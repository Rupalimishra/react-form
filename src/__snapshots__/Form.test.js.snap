// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <div>
    <div
      className="form-container"
    >
      <FormSignup
        submitForm={[Function]}
      />
    </div>
  </div>,
]
`;
